src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+[assembly: System.Resources.NeutralResourcesLanguage("en-us")]
+
+namespace Microsoft.Extensions.Logging.Generators
+{
+    [Generator]
+    public partial class LoggerMessageGenerator : ISourceGenerator
+    {
+        [ExcludeFromCodeCoverage]
+        public void Initialize(GeneratorInitializationContext context)
+        {
+            context.RegisterForSyntaxNotifications(SyntaxReceiver.Create);
+        }
+
+        [ExcludeFromCodeCoverage]
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var receiver = context.SyntaxReceiver as SyntaxReceiver;
+            if (receiver == null || receiver.ClassDeclarations.Count == 0)
```

Comment:
`if (context.SyntaxReceiver is SyntaxReceiver receiver || receiver.ClassDeclarations.Count == 0)`

Reply:
old school:
!(context.SyntaxReceiver is SyntaxReceiver receiver) 

new school:
context.SyntaxReceiver is not SyntaxReceiver receiver

Reply:
`!(context.SyntaxReceiver is SyntaxReceiver receiver)` is probably "middle school" since you couldn't do that in C# 6.

It takes me a while to embed new language features in my brain.